Tests/MuxUploadSDKTests/Upload Tests/ReporterTests.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ReporterTests.swift
+//
+
+import Foundation
+import XCTest
+
+@testable import MuxUploadSDK
+
+class ReporterTests: XCTestCase {
+
+    func testUploadEventSerialization() throws {
+        let reporter = Reporter()
+        reporter.pendingUploadEvent = UploadEvent(
+            startTime: 100,
+            endTime: 103,
+            fileSize: 1_500_000,
+            videoDuration: 3.14,
+            uploadURL: URL(string: "https://www.example.com")!,
+            sdkVersion: "0.3.1",
+            osName: "iPadOS",
+            osVersion: "16.2",
+            deviceModel: "iPad",
+            appName: "foo",
+            appVersion: "14.3.1",
+            regionCode: "US"
+        )
+
+        let serializedUploadEvent = try reporter.serializePendingEvent()
+
+        let json = try XCTUnwrap(
+            String(
+                data: serializedUploadEvent,
+                encoding: .utf8
+            )
+        )
+
+        XCTAssertEqual(
```

Comment:
Just a suggestion: you might be able to avoid maintaining this big string by re-serializing the json then comparing that against the original `UploadEvent`. Your `JSONDecoder` config would have to match the invisible `JSONEncoder`'s, but it still seems easier to deal with than the json string here

wdyt?

Reply:
I think testing the decoded event makes sense if the test needs to validate round trip serialization and deserialization of `UploadEvent`. 

Since the SDK implementation never actually needs to decode this event and the goal is to keep reported data as consistent as possible across many SDK versions, it is intentional that the test fails if the properties get renamed or a JSON key is added / removed, as these are noteworthy changes.

Reply:
This is a good callout though, as more events get added then there does need to be some sort of scheme in place to keep the raw strings organized.

Reply:
fair enough! Thanks for the discussion

Reply:
As an example of what I mean here is one approach that builds raw JSON out of substrings, can be made more elaborate if needed: https://github.com/apple/swift-protobuf/blob/main/Tests/SwiftProtobufTests/Test_JSON.swift